lib/components/base-components/NormalComponent/NormalComponent.ts

```diff
@@ -375,6 +376,22 @@ export class NormalComponent<
     return s.startsWith("http://") || s.startsWith("https://")
   }
 
+  /**
+   * Returns the library prefix and footprintName̦
+   * reference - "<lib>:<footprintName>"
+   */
+  _parseLibraryFootprintRef(
+    s: string,
+  ): { lib: string; footprintName: string } | null {
+    if (this._isFootprintUrl(s)) return null
+    const idx = s.indexOf(":")
+    if (idx <= 0) return null
+    const lib = s.slice(0, idx)
+    const footprintName = s.slice(idx + 1)
+    if (!lib || !footprintName) return null
+    return { lib, footprintName }
+  }
```

Comment:
this doesn't need to be a class method, similar to `isFootprintUrl`, we should generally avoid class methods which are hard to refactor. You can just move this to a file, even in the NormalComponent directory is fine. `parseLibraryFootprintRef` and `isFootprintUrl`. It will help reduce the size of this file